src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -576,3 +611,513 @@ void JasmineGraphFrontEnd::removeGraph(std::string graphID, void *dummyPt) {
+            } else {
+                aggregatorPartitionId = partitionId;
+            }
+
+        }
+
+    }
+
+
+    for (auto&& futureCall:intermRes) {
+        result += futureCall.get();
+    }
+
+
+
+    long aggregatedTriangleCount = JasmineGraphFrontEnd::countCentralStoreTriangles(aggregatorWorkerHost,aggregatorWorkerPort,aggregatorWorker,aggregatorPartitionId,graphId);
+    result += aggregatedTriangleCount;
+    return result;
+}
+
+
+long JasmineGraphFrontEnd::getTriangleCount(int graphId, std::string host, int port, int partitionId) {
+
+    int sockfd;
+    char data[300];
+    bool loop = false;
+    socklen_t len;
+    struct sockaddr_in serv_addr;
+    struct hostent *server;
+    Utils utils;
+    long triangleCount;
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+
+    if (sockfd < 0) {
+        std::cerr << "Cannot accept connection" << std::endl;
+        return 0;
+    }
+
+    if (host.find('@') != std::string::npos) {
+        host = utils.split(host, '@')[0];
+    }
+
+    server = gethostbyname(host.c_str());
+    if (server == NULL) {
+        std::cerr << "ERROR, no host named " << server << std::endl;
+    }
+
+    bzero((char *) &serv_addr, sizeof(serv_addr));
+    serv_addr.sin_family = AF_INET;
+    bcopy((char *) server->h_addr,
+          (char *) &serv_addr.sin_addr.s_addr,
+          server->h_length);
+    serv_addr.sin_port = htons(port);
+    if (connect(sockfd, (struct sockaddr *) &serv_addr, sizeof(serv_addr)) < 0) {
+        std::cerr << "ERROR connecting" << std::endl;
+        //TODO::exit
+    }
+
+    bzero(data, 301);
+    write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(), JasmineGraphInstanceProtocol::HANDSHAKE.size());
+    frontend_logger.log("Sent : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
+    bzero(data, 301);
+    read(sockfd, data, 300);
+    string response = (data);
+
+    response = utils.trim_copy(response, " \f\n\r\t\v");
+
+    if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) == 0) {
+        frontend_logger.log("Received : " + JasmineGraphInstanceProtocol::HANDSHAKE_OK, "info");
+        string server_host = utils.getJasmineGraphProperty("org.jasminegraph.server.host");
+        write(sockfd, server_host.c_str(), server_host.size());
+        frontend_logger.log("Sent : " + server_host, "info");
+
+        write(sockfd, JasmineGraphInstanceProtocol::TRIANGLES.c_str(),
+              JasmineGraphInstanceProtocol::TRIANGLES.size());
+        frontend_logger.log("Sent : " + JasmineGraphInstanceProtocol::TRIANGLES, "info");
+        bzero(data, 301);
+        read(sockfd, data, 300);
+        response = (data);
+        response = utils.trim_copy(response, " \f\n\r\t\v");
+
+        if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
```

Comment:
Why do we have the same if check duplicated in line 771?

Reply:
It is due to the fact that the input parameters are sent one by one after initiating the triangle count algorithm. For example the graph id and partition id will be sent as two input parameters as subsequent inputs. Hence if one input is accept and valid only the other input will be captured. For that we have to use two if statements